metaflow/extension_support.py

```diff
@@ -0,0 +1,441 @@
+
+    try:
+        with open(INFO_FILE, "r") as contents:
+            all_pkg, ext_to_pkg = json.load(contents).get("ext_info", (None, None))
+            if all_pkg is not None and ext_to_pkg is not None:
+                # We need to properly convert stuff in ext_to_pkg
+                for k, v in ext_to_pkg.items():
+                    v = [MFExtPackage(*d) for d in v]
+                    ext_to_pkg[k] = v
+                return all_pkg, ext_to_pkg
+    except IOError:
+        pass
+
+    _list_ext_points = [x.split(".") for x in _extension_points]
+    # TODO: This relies only on requirements to determine import order; we may want
+    # other ways of specifying "load me after this if it exists" without depending on
+    # the package. One way would be to rely on the description and have that info there.
+    # Not sure of the use though so maybe we can skip for now.
+    mf_ext_packages = []
+    extension_points_to_pkg = defaultdict(dict)
+    for dist in metadata.distributions():
+        if any(
+            [pkg == EXT_PKG for pkg in (dist.read_text("top_level.txt") or "").split()]
+        ):
+            _ext_debug("Found extension package '%s'..." % dist.metadata["Name"])
+            mf_ext_packages.append(dist.metadata["Name"])
+            # At this point, we check to see what extension points this package
+            # contributes to. This is to enable multiple namespace packages to contribute
+            # to the same extension point (for example, you may have multiple packages
+            # that have plugins)
+            for f in dist.files:
+                parts = list(f.parts)
+                if len(parts) > 3 and parts[0] == EXT_PKG:
+                    # We go over _extension_points *in order* to make sure we get more
+                    # specific paths first
+                    for idx, ext_list in enumerate(_list_ext_points):
+                        if (
+                            len(parts) > len(ext_list) + 2
+                            and parts[2 : 2 + len(ext_list)] == ext_list
+                        ):
+                            # Check if this is an "init" file
+                            config_module = None
+                            if (
+                                len(parts) == len(ext_list) + 3
+                                and EXT_CONFIG_REGEXP.match(parts[-1]) is not None
+                            ):
+                                parts[-1] = parts[-1][:-3]  # Remove the .py
+                                config_module = ".".join(parts)
+
+                            cur_pkg = extension_points_to_pkg[
+                                _extension_points[idx]
+                            ].get(dist.metadata["Name"], None)
+                            if cur_pkg is not None:
+                                if (
+                                    config_module is not None
+                                    and cur_pkg.config_module is not None
```

Comment:
Can you elaborate about why we do this?

Reply:
This is just to catch the case of multiple config files.